src/Client/ClientBase.cpp

```diff
@@ -525,16 +525,26 @@ try
                 out_file = out_file_node.value.safeGet<std::string>();
 
                 std::string compression_method;
+                UInt64 compression_level = 3;
                 if (query_with_output->compression)
                 {
                     const auto & compression_method_node = query_with_output->compression->as<ASTLiteral &>();
                     compression_method = compression_method_node.value.safeGet<std::string>();
+
+                    if (query_with_output->compression_level)
+                    {
+                        const auto & compression_level_node = query_with_output->compression_level->as<ASTLiteral &>();
+                        bool res = compression_level_node.value.tryGet<UInt64>(compression_level);
+
+                        if (!res || compression_level < 1 || compression_level > 9)
```

Comment:
Why only from 1 to 9? E.g. lz4 support compression levels from 1 to 12 and zstd from 1 to 22.